libs/sso-client/src/lib/sso-client.ts

```diff
@@ -32,21 +32,29 @@ export const middlewareHandler =
     exhaustive = true
   ) =>
   async (request: NextRequest): Promise<NextResponse | null> => {
-    const access_token_init = request.cookies.get(
-      getEnv().Hibiscus.Cookies.accessTokenName
-    );
-
-    if (getEnv().Hibiscus.Cookies.disableSSO === 'true') {
-      await initializeFakeUser(request);
-    }
-
     // Handle preflight requests
     if (request.method === 'OPTIONS') {
       const res = NextResponse.next();
       res.headers.set('Access-Control-Allow-Credentials', 'true');
       return res;
     }
 
+    let access_token = request.cookies.get(
+      getEnv().Hibiscus.Cookies.accessTokenName
+    );
+    let refresh_token = request.cookies.get(
+      getEnv().Hibiscus.Cookies.refreshTokenName
+    );
+
+    const access_token_init = access_token;
+
+    if (getEnv().Hibiscus.Cookies.disableSSO === 'true') {
```

Comment:
could `disableSSO` be a boolean considering it's only either `'true'` or `'false'`?

Reply:
I think env vars are always strings so it should be safer to check for the actual string

Reply:
I think the env library is supposed to provide a type-safe ref to env variables parsed at runtime but it should also interpret env vars e.g `DEBUG=1` means debug is true; keeping it as string like this is a bit unintuitive and loses the meaning of it since people would expect `disableSso` to be a boolean due to its name; leaving it as a string makes it a bit ambiguous to infer e.g another person trying to see if SSO is disabled or not, should they check if its equal to `1` or `true`?

Reply:
Changed!